### PR TITLE
Make use of deterministic guids

### PR DESCRIPTION
**Pull request by jmacato**
@kekekeks @danwalmsley

### CODE REVIEW

File: src/XamlX/Helpers/DeterministicName.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
+﻿using System.Text;
+using System.Linq;
+using System.Threading;
+using System.Security.Cryptography;
+using System;
+
+namespace XamlX.Helpers
+{
+    public class DeterministicName
```
Comment (kekekeks) [inline src/XamlX/Helpers/DeterministicName.cs]: MSBuild task assembly is reused across builds, so your counter will stay in memory.

File: src/XamlX/Ast/Clr.cs [hunk 2]
```diff
@@ -640,7 +641,7 @@ public XamlILNodeEmitResult Emit(XamlEmitContext<IXamlILEmitter, XamlILNodeEmitR
         {
             var so = context.Configuration.WellKnownTypes.Object;
             var isp = context.Configuration.TypeMappings.ServiceProvider;
-            var subType = context.CreateSubType("XamlClosure_" + Guid.NewGuid(), so);
+            var subType = context.CreateSubType("XamlClosure_" + DeterministicName.GetString(), so);
```
Comment (kekekeks) [inline src/XamlX/Ast/Clr.cs]: That is the ONLY relevant change in the entire PR. Unfortunately, your DeterministicName implementation is not deterministic.

Comment (danwalmsley) [reply src/XamlX/Ast/Clr.cs]: Brutal!

**Comment by jmacato** [comment]
Closing this in favor of https://github.com/kekekeks/XamlX/pull/19

**Pull request closed without merging**